### PR TITLE
Remove public node settings from cli

### PR DESCRIPTION
**Pull request by sorpaas**
rel #8649.

Remove public node related settings. Keep legacy `--public-node` args and `public_node` item in config.toml to keep it backward compatible.

### CODE REVIEW

File: parity/run.rs [hunk 1]
```diff
@@ -721,11 +720,7 @@ fn execute_impl<Cr, Rr>(cmd: RunCmd, logger: Arc<RotatingLogger>, on_client_rq:
 
 	// set up dependencies for rpc servers
 	let rpc_stats = Arc::new(informant::RpcStats::default());
-	let secret_store = match cmd.public_node {
-		true => None,
-		false => Some(account_provider.clone())
-	};
-
+	let secret_store = Some(account_provider.clone());
```
Comment (ascjones) [inline parity/run.rs]: Is it possible now to remove the `Option` from `secret_store` in the rpc dependencies? Then could remove these unwraps e.g. https://github.com/paritytech/parity/blob/master/rpc/src/v1/impls/signing_unsafe.rs#L111.

**Comment by ascjones** [comment]
Just some compilation errors in some mocks

**Comment by 5chdn** [comment]
Question: is there anything left from public node after this is merged?

**Comment by sorpaas** [comment]
@5chdn This remove all things enabled by the `--public-node` flag. So I think there shouldn't be anything else left for public node.

**Pull request merged**